catalog/app/containers/Admin/Table/typed.tsx

```diff
@@ -95,73 +163,14 @@ export function Toolbar({
         )}
       </div>
       <div className={classes.spacer} />
-      {children}
+      {ordering?.filter && <Filter {...ordering.filter} />}
```

Comment:
i feel very strongly against such "abstracting" in favor of simple composition. like, if we have a filtering model exposed by the hypothetical `useFiltering` hook, we can just render this Filter component explicitly using that model when we need it, and not overload the generalized Toolbar component with very specific functionality which can be easily "composed" instead

Reply:
Or, this filter can be in any Table and be part of both Table and Table.useOrdering?

Reply:
Maybe, by "order" you mean "sort" only?

Reply:
@nl0 ping

Reply:
the original intent behind that design was to make individual pieces composable and not inter-dependent whenever possible. 

> Maybe, by "order" you mean "sort" only?

well, these terms are almost interchangeable in the current context, and filtering is orthogonal to order/sorting

Reply:
So, can I rename it to `useSorting`? Because in my head `useOrdering` means to keep items in order: sort and filter them.

Reply:
well, the words "order" / "ordering" feel natural to me to describe the concept of, well, elements (i.e. of a set) having a particular order, following one another, while "sorting" being a specific operation of sorting a list or smth (maybe because of me thinking in terms of functional languages where "Order" quite often is a literal name of a typeclass providing specific properties related to ordering things). it correlates well with the often used phrase "order by". but in the end ordering feels distinct from filtering. order doesn't imply including / excluding elements, only their sequence. filtering is orthogonal to ordering.